lib/commands/backup.js

```diff
@@ -0,0 +1,178 @@
+
+function absolutePath(inputPath) {
+    return path.isAbsolute(inputPath) ? inputPath : path.join(process.cwd(), inputPath);
+}
+
+class BackupCommand extends Command {
+    run(argv) {
+        return this.ui.listr([{task: this.initialize.bind(this),
+            title: 'Checking configuration'}],{argv: argv})
+            .then(() => {
+                return this.ui.listr([{
+                    title: 'Backing up extensions',
+                    task: this.backupExtensions.bind(this)
+                }, {
+                    title: 'Backing up database',
+                    task: this.backupDatabase.bind(this)
+                }, {
+                    title: 'Backing up config files',
+                    task: this.backupConfig.bind(this)
+                }, {
+                    title: 'Backing up content folder',
+                    task: this.backupContent.bind(this)
+                }, {
+                    title: `Writing backup file to ${this.saveLocation}`,
+                    task: this.writeZipFile.bind(this)
+                }]);
+            });
+    }
+
+    initialize(ctx) {
+        const instance = this.system.getInstance();
+        const datetime = (new Date()).toJSON().substring(0, 10);
+
+        this.database = {};
+        this.saveLocation = process.cwd();
+
+        if (ctx.argv.output) {
+            this.saveLocation = absolutePath(ctx.argv.output);
+        }
+
+        try {
+            fs.ensureDirSync(this.saveLocation);
+            // After we're sure the directory exists, check to make sure we can write to it
+            fs.access(this.saveLocation, fs.W_OK);
+        } catch (error) {
+            this.ui.log('Ghost doesn\'t have permission to write to the output folder', 'red');
+            return Promise.reject(new cli.errors.SystemError(error));
+        }
+
+        this.saveLocation = path.join(this.saveLocation, `${instance.name}.backup.${datetime}.zip`);
+
+        // The zip library used (adm-zip) has a known issue where the zip is structurally sound,
+        // but Windows has issues reading / unzipping it. To counteract this, all files are read
+        // and added via the `addFile` method, where the file permissions can be set, seemingly
+        // resolving the issue. See https://github.com/cthackers/adm-zip/issues/171
+        this.zipFile = new Zip();
+
+        if (instance.running()) {
+            this.ui.log('Ghost is currently running. Backing up might take longer and slow down your site', 'yellow');
+            instance.loadRunningEnvironment();
+        } else {
+            instance.checkEnvironment();
+        }
+
+        return Promise.resolve();
+    }
+
+    backupContent() {
+        let files;
+        try {
+            files = walker('./content', {nodir: true});
+        } catch (E) {
+            debug(`Failed reading content folder: ${E}`);
+            return Promise.reject(new Error('Failed to read content folder'));
```

Comment:
This should probably be a SystemError rather than a regular error, mainly because regular errors will output a stack trace into the log file, and we don't necessarily need one here.